bitrise-io/bitrise
# Tool 1347 log new step version info (#692)

* refactor: extract function

* print resolved version when applicable

* refactor: simplify conditionals

* refactor: simplify conditional

* test cases: version row for major and minor lock

* build correct version info row string

* refactor: remove useless code

* print dotted changelog URL

* refactor: put major and minor test case together

* test case: url cropping

* empty line in box under version info

* dont show update info if latest version fits
lock

* remove development related temp code

* remove unrelated logic from tests

* improve test case name

* refactor: simplify string production

* test case for 1.x and 1.x.x format

* refactor: extract update handling to separate file

* refactor: consolidate update check flow

* handle semver parse error

* remove testing code

* refactor: make naming consistent

* monkey patch dependency

* fallback to repo url when releases url is unknown

* dep update
